tests/phpunit/tests/html-api/wpHtmlProcessorState.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Processor_State functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ *
+ * @since 6.7.0
+ *
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor_State
+ */
+class Tests_HtmlApi_WpHtmlProcessorState extends WP_UnitTestCase {
+	/**
+	 * @dataProvider data_insertion_mode_cases
+	 *
+	 * @ticket 61549
+	 *
+	 * @param array $stack_of_open_elements   Stack of open elements.
+	 * @param string $expected_insertion_mode Expected insertion mode after running the algorithm.
+	 */
+	public function test_reset_insertion_mode(
+		array $stack_of_open_elements,
+		string $expected_insertion_mode
+	): void {
+		$state               = new WP_HTML_Processor_State();
+		$state->context_node = array( 'BODY', array() );
+
+		foreach ( $stack_of_open_elements as $i => $tag_name ) {
+			if ( ! ctype_upper( $tag_name ) ) {
+				throw new Error( 'Expected upper case tag names.' );
```

Comment:
I don't like this error message. for one, it's testing the wrong thing in the wrong test. for two, it doesn't help explain why this test is failing.

what's the purpose of this? is it to get around markers and text nodes and comments, etc…?